pkg/prm/config.go

```diff
@@ -49,14 +49,19 @@ func GenerateDefaultCfg() {
 }
 
 func LoadConfig() error {
-	// Load Puppet version from config
-	pupperSemVer, err := semver.NewVersion(viper.GetString(PuppetVerCfgKey))
+	// If the scenario where any other config value has been set AND the Puppet version is unset, a '{}' is written
+	// to the config file on disk. This causes issues when attempting to call semver.NewVersion.
+	puppetVer := viper.GetString(PuppetVerCfgKey)
+	if puppetVer == "" {
+		puppetVer = DefaultPuppetVer
+	}
+	pupperSemVer, err := semver.NewVersion(puppetVer)
 	if err != nil {
-		return fmt.Errorf("could not load '%s' from config '%s': %s", PuppetVerCfgKey, viper.GetViper().ConfigFileUsed(), err)
+		log.Error().Msgf("could not load '%s' from config '%s': %s", PuppetVerCfgKey, viper.GetViper().ConfigFileUsed(), err)
```

Comment:
Will the error need to be returned in this conditional?

Reply:
Yes, indeed it should. I initially had a different method that was going to continue and missed this when I changed! Good catch!